incubator/java-openliberty/image/Dockerfile-stack

```diff
@@ -34,17 +34,17 @@ ENV APPSODY_USER_RUN_AS_LOCAL=true
 # Set APPSODY_DEV_MODE to get in the habit in case we need it someday
 ENV APPSODY_PREP="export APPSODY_DEV_MODE=prep;  ../validate.sh"
 
-ENV APPSODY_RUN="export APPSODY_DEV_MODE=run; mvn -B -DinstallDirectory=/opt/ol/wlp -Dmaven.repo.local=/mvn/repository liberty:dev"
+ENV APPSODY_RUN="export APPSODY_DEV_MODE=run; mvn -B -DappsDirectory=apps -DinstallDirectory=/opt/ol/wlp -Dmaven.repo.local=/mvn/repository liberty:dev"
```

Comment:
I know this is in part a workaround for [this](https://github.com/OpenLiberty/ci.maven/issues/718).    Might be cleaner to just add <appsDirectory> config once in POM rather than in each mvn invocation.  Also if it were nicer to put in a template POM rather than the stack POM, then you require each template to know about this .. which argues for putting in the stack POM (following my idea).    Is that a good fit?   Well, that also suggests we could use some more doc on how to navigate through the plugin config in each POM/mode.

Reply:
I also don't see why we need to include the `-DinstallDirectory=/opt/ol/wlp`.     Can we try collapsing the stack package path and the `appsody run/debug/test` paths back together (but leaving distinct from the host-only path)?   I think we either wrongly reacted to some other bug or are dealing with docker timestamp lag (and if the latter, I'd like to raise this to the ci.maven project).

Reply:
1. The `appsDirectory` property will need to be set in the child pom. If we set it in the parent pom, it would be set in the stack-package profile which isnt invoked when liberty:dev is called on the child. Since this is a temporary workaround (and only a problem when we are in the docker / appsody run flow), I think we should leave it here rather than muddling with the child pom.  

2. I dont see a clean way to collapse these together. The idea (I believe) would be to have the child pom inherit the `runtimeInstallDirectory` property from the parent. Last time we tested this, I believe the profile definitions were a bit different, but now, the child can't inherit this property for the same reason above^. The stack-package profile is not active when the child pom builds. But even if it were, it would then just expose the potential timestamp bug that leads to uninstalling the runtime. Setting the `installDirectory` property on the child pom tells the LMP that the runtime is _already_ installed rather than telling it a location _to install_ it (or potentially reinstall it). If this is a bug in ci.maven, once fixed we can just remove the -DinstallDirectory property from the maven invocations just like we can remove the -DappsDirectory property once that bug is addressed.